### PR TITLE
Pd 3164 cbo cbolist filter does not persist

### PR DESCRIPTION
**Pull request by aljordan82**
## What's this PR do?
    Implement Redux to store the CBO filter state.
    Use bootstrap table to display CBO list

### CODE REVIEW

**Comment by mwarnsleyDE** [comment]
@aljordan82 I put a DO NOT MERGE on this just so we can merge the rest of what we have and have you do a pull to make sure there aren't any conflicts which I don't think there should be. As well, I am guessing if it's easier for us to use the bootstrap table that we should persist it which would mean implementing it inside of the ContactList.js
DistributionLocationList.js and RulesList.js where there are tables as well.

**Comment by mwarnsleyDE** [comment]
@aljordan82 so there are going to be some merge conflicts that you'll have to fix since a lot of that stuff got merged into Quality-Control

**Comment by koonsies** [comment]
Can we change "City" in the filter menu to "Distribution City"? To match "Distribution State"

File: gulp/src/cbo/actions/cboActions.js [hunk 1]
```diff
@@ -266,6 +271,105 @@ export function deleteDistributionLocation(zipperedCbo) {
...
+      if (a.isCustom > b.isCustom) {
+        return 1;
+      }
+      return 0;
+    };
+
+    cbos.sort(sortFunc);
+    function searchCategoryForValue(contact) {
+      return contact.category && contact.category.some((category) => {
+        return this(category);
```
Comment (koonsies) [inline gulp/src/cbo/actions/cboActions.js]: I'm not familiar with this use of "this". What does this line do?

Comment (aljordan82) [inline gulp/src/cbo/actions/cboActions.js]: 'this' is the matcher function "categoryMatcher"

Comment (koonsies) [inline gulp/src/cbo/actions/cboActions.js]: Interesting. I admit I haven't done a ton with context binding, so using "this" like that is pretty new to me haha.

**Pull request merged**